### PR TITLE
docs: improves CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by kevin-kline**
Rephrasing, spelling, capitalization, "closing issues" section, "approval of issues" section,
"proposing a change" section

### CODE REVIEW

**Comment by UltimateBeaver** [comment]
@kevin-kline The current version reads a bit "internal-facing". I suggest we go through the document again from the perspective of an outside contributor. For example, the SLA should only apply to our Merico colleagues. 

I found a really good [CONTRIBUTING.md(https://github.com/atom/atom/blob/master/CONTRIBUTING.md) by atom, feel free to refer to it and even reuse paragraphs, just to make sure we give attribution to them at the bottom of the doc.

Their first paragraph is really welcoming, I suggest we reuse:

```
👍🎉 First off, thanks for taking the time to contribute! 🎉👍

The following is a set of guidelines for contributing to Atom and its packages, which are hosted in the Atom Organization on GitHub. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
```

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,26 +1,40 @@
...
-## Run lake as a developer
+## Collaboration Guidelines
 
-```
-npm i
-cp config/local.sample.js config/local.js
-npm run compose-dev
-npm run lake
-```
+1. Good collaboration relies on high-quality asynchronous communication (Issues, PRs, and READMEs). Writing down our thoughts as clearly and specifically as possible may slow us down in the short term, but will definitely benefit us all in the long term.
```
Comment (UltimateBeaver) [inline CONTRIBUTING.md:5]: Let's remove this paragraph. It was meant for internal synchronization, most open-source developers wouldn't need this paragraph.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+
+2. If you'd like to start working on a new feature, bug, or change of any kind, please open a new Issue in GitHub with the "proposal" label before you begin work. Describe your proposal in great detail. You may use code samples. Invite the related colleagues to discuss and iterate on your proposal. Once it's approved, you can start implementing it and remove the proposal label, and add "in progress" label. This is our mechanism for "acquiring the lock".
+
+3. If you're mentioned in an Issue, the default SLA (Service Level Agreement) to reply is 24 hours. If you're busy that day, no worries. Simply reply and explain the situation. A timely reply to Issues will encourage our colleagues to communicate via high-quality Issues.
```
Comment (UltimateBeaver) [inline CONTRIBUTING.md:9]: Let's rephrase this as "The maintainer team aims to achieve an SLA of 24 hrs for replying to issues...". This is not a requirement for the general contributors.

**Comment by hezyin** [comment]
LGTM
